src/olympia/amo/middleware.py

```diff
@@ -188,15 +191,50 @@ def process_request(self, request):
 
 
 class ReadOnlyMiddleware(object):
+    """Middleware that announces a downtime which for us usually means
+    putting the site into read only mode.
+
+    Supports issuing `Retry-After` header.
+    """
+    ERROR_MSG = _(
+        u'Some features are temporarily disabled while we '
+        u'perform website maintenance. We\'ll be back to '
+        u'full capacity shortly.')
+
+    API_HEADER_NAME = 'X-AMO-Read-Only'
```

Comment:
nit: understand you are trying to keep it generic, but calling it something more descriptive makes the code below easier to follow, imo.  Something like `READ_ONLY_HEADER`

Reply:
Fixed in 63015a01a6